daemons/ipa-slapi-plugins/ipa-extdom-extop/ipa_extdom_common.c

```diff
@@ -97,25 +98,48 @@ static int inc_buffer(size_t buf_max, size_t *_buf_len, char **_buf)
     return 0;
 }
 
-int getpwnam_r_wrapper(size_t buf_max, const char *name,
+int __nss_to_err(enum nss_status errcode)
+{
+    int ret = -1;
+
+    if (errcode == NSS_STATUS_SUCCESS) {
+        ret = 0;
+    } else if (errcode == NSS_STATUS_NOTFOUND) {
+        ret = ENOENT;
+    } else if(errcode == NSS_STATUS_TRYAGAIN) {
+        ret = ERANGE;
```

Comment:
This mapping seem wrong, why ERANGE ?
This sounds more like EAGAIN

Reply:
Because it is handled this way on nsswitch side. ERANGE is returned when insufficient buffer is supplied. There is no EAGAIN returned -- see getgrnam(3) or getpwnam(3).

Reply:
ok makes sense